server/src/termdb.regression.js

```diff
@@ -1201,35 +1205,27 @@ async function getSampleData_dictionaryTerms(q, terms) {
 	)
 	values.push(...terms.map(d => d.term.id || d.term.name))
 
-	const sql = `WITH
-		${filter ? filter.filters + ',' : ''}
-		${CTEs.map(t => t.sql).join(',\n')}
-		${CTEs.map(
-			t => `
-			SELECT sample, key, value, ? as term_id
-			FROM ${t.tablename}
-			${filter ? `WHERE sample IN ${filter.CTEname}` : ''}
-			`
-		).join(`UNION ALL`)}`
-
-	const _rows = q.ds.cohort.db.connection.prepare(sql).all(values)
+	const onlyChildren = sample_types.size > 1
+	const _rows = await getAnnotationRows(q, terms, filter, CTEs, values, onlyChildren)
 
 	// process rows
 	const rows =
 		q.regressionType == 'cox' && q.outcome.term.type == 'condition'
 			? processCoxConditionOutcomeRows(_rows, q.outcome, q.ds.cohort.termdb.ageEndOffset)
 			: _rows
-
 	// parse the processed rows
+
 	for (const { sample, term_id, key, value } of rows) {
-		const term = terms.find(term => (term.term.id || term.term.name) == term_id)
+		addSample(sample, term_id, key, value)
+	}
 
+	function addSample(sample, term_id, key, value) {
+		const term = terms.find(term => term.term.id || term.term.name == term_id)
```

Comment:
I think `term.term.id || term.term.name == term_id` should be `(term.term.id || term.term.name) == term_id` to resolve `term.term.id || term.term.name` before making the comparison